src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -549,7 +549,14 @@ private BoundStatement BindLocalDeclarationStatement(LocalDeclarationStatementSy
             var binder = GetBinder(node);
             Debug.Assert(binder != null);
 
-            return binder.WrapWithVariablesIfAny(node, binder.BindDeclarationStatementParts(node, diagnostics));
+            if (node.Declaration.Deconstruction == null)
+            {
+                return binder.WrapWithVariablesIfAny(node, binder.BindDeclarationStatementParts(node, diagnostics));
+            }
+            else
+            {
+                return binder.WrapWithVariablesIfAny(node, binder.BindDeconstructionDeclarationStatementParts(node, diagnostics));
+            }
```

Comment:
Consider moving condition inside `binder.WrapWithVariablesIfAny(...)` or move `binder.WrapWithVariablesIfAny()` outside of `if`.